### PR TITLE
fix: Button icon order and margin in rtl mode

### LINKED ISSUE #48807: Broken icon margin in RTL button
### Reproduction link

[https://ant.design/components/button?direction=rtl](https://ant.design/components/button?direction=rtl)

### Steps to reproduce

Add icon to RTL button.

### What is expected?

Content of the button is in the middle with space between text and icon.

### What is actually happening?

Icon in RTL button with iconPosition="start" has no margin from the text.
Icon in RTL button with iconPosition="end" has unnecessary margin which makes content of the button not in the middle.

| Environment | Info |
| --- | --- |
| antd | 5.17.0 |
| React | 18.3.1 |
| System | MacOS 14.4.1 |
| Browser | Edge 124.0.2478.80 |

### PR DESCRIPTION
**Pull request by afc163**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix https://github.com/ant-design/ant-design/issues/48807
close https://github.com/ant-design/ant-design/pull/48813
close https://github.com/ant-design/ant-design/pull/48811

### 💡 Background and solution

RTL 本身会改变文本顺序，不需要增加额外判断。https://github.com/ant-design/ant-design/pull/47791/files#r1593303396

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Button icon wrong order and margin in RTL mode.          |
| 🇨🇳 Chinese |  修复 Button 在 RTL 下的图标顺序和边距缺失问题。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/button/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 1]
```diff
@@ -2774,7 +2774,11 @@ exports[`renders components/button/demo/multiple.tsx extend context correctly 1`
 </div>
 `;
 
-exports[`renders components/button/demo/multiple.tsx extend context correctly 2`] = `[]`;
+exports[`renders components/button/demo/multiple.tsx extend context correctly 2`] = `
+[
+  "Warning: findDOMNode is deprecated and will be removed in the next major release. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node%s",
+]
```
Comment (afc163) [inline components/button/__tests__/__snapshots__/demo-extend.test.ts.snap:2780]: 为啥这个会变？。。

Comment (crazyair) [reply components/button/__tests__/__snapshots__/demo-extend.test.ts.snap:2780]: 这是警告吧，都打到快照了。。。

File: components/button/button.tsx [hunk 2]
```diff
@@ -258,16 +258,18 @@ const InternalCompoundedButton = React.forwardRef<
...
       <>
-        {iconFirst ? iconComponent : kidsComponent}
-        {iconFirst ? kidsComponent : iconComponent}
+        {iconComponent}
+        {kidsComponent}
+      </>
+    ) : (
+      <>
+        {kidsComponent}
+        {iconComponent}
```
Comment (crazyair) [inline components/button/button.tsx:270]: 用 reverse? 

```tsx
cosnt list=[iconComponent ,kidsComponent]
{iconPosition === 'start'? list : list.reverse()}
```

Comment (afc163) [reply components/button/button.tsx:270]: 数组要补 key，比较麻烦。

File: components/button/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 3]
```diff
@@ -2774,7 +2774,11 @@ exports[`renders components/button/demo/multiple.tsx extend context correctly 1`
...
+`;
```
Comment (afc163) [inline components/button/__tests__/__snapshots__/demo-extend.test.ts.snap:2781]:
```suggestion
exports[`renders components/button/demo/multiple.tsx extend context correctly 2`] = `[]`;
```

**Comment by Wxh16144** [comment]
https://github.com/ant-design/ant-design/pull/48813/commits/68227f82614cfc11f55fbf7ed9f4c995ecddaf65 这个要不要看一下，感觉可以 revert。 原因我在我的那个 pr 里面描述了一嘴

File: components/button/button.tsx (hunk 2 above)
Comment (Wxh16144) [reply components/button/button.tsx:270]: 我昨天也注意到这个写法，看起来很奇怪，但是用 reverse 会有 key 的错误。 当初我的写法是：

```tsx

const nodes = [
  <React.Fragment key="icon">{iconComponent}</React.Fragment>,
  <React.Fragment key="content">{kidsComponent}</React.Fragment>,
]
return iconPosition === 'start' ? nodes : nodes.reverse();

```

**Comment by afc163** [comment]
@Wxh16144 对，感觉也可以回滚。

**Comment by afc163** [comment]
想合并了 https://github.com/ant-design/ant-design/pull/47318 后用 `flex-direction: row-reverse;` 来做 `iconPosition`，代码会简单很多。

**Review by kiner-tang — APPROVED** [review]

**Pull request merged**